OneSignalSDK/onesignal/build.gradle

```diff
@@ -61,9 +61,10 @@ dependencies {
     // Also note, firebase & gms libraries use android.support:26.
     //   - They never refer to 27 or 28
     api 'com.android.support:cardview-v7:[26.0.0, 27.99.99]'
-    api 'com.android.support:support-v4:[26.0.0, 27.99.99]'
```

Comment:
Even if we remove `com.android.support:support-v4` from here it is still being included through other dependencies.

```
+--- com.google.firebase:firebase-messaging:[10.2.1, 17.3.99] -> 17.3.4
|    +--- com.google.android.gms:play-services-basement:16.0.1
|    |    \--- com.android.support:support-v4:26.1.0
```


By adding the following to your build.gradle you will observe errors in the OneSignal code base due to force removing this dependency.
```
configurations.all {
    exclude group: 'com.android.support', module: 'support-v4'
}
```

Reply:
It looks like however we might be able to get away with ` api 'com.android.support:support-fragment:` instead if we want to switch to that. It seems to include everything we need.

Reply:
One last note with dependencies, any OneSignal code that directly uses a library it should be top level in this file as the fundamentally matches up.